src/os.c

```diff
@@ -237,13 +242,27 @@ void _mi_os_init() {
 
 #else  // generic unix
 
+static void os_detect_transparent_huge_pages(void) {
+#if defined(MADV_HUGEPAGE)
+  int fd = open("/sys/kernel/mm/transparent_hugepage/enabled", O_RDONLY);
+  if (fd < 0) return;
+  char buf[128];
+  ssize_t nread = read(fd, &buf, sizeof(buf));
+  close(fd);
+  if (nread >= 1) {
+      // in some configurations, it can occur (e.g. embedded)
+      if (!strstr(buf, "[never]")) os_transparent_huge_pages = true;
```

Comment:
What are the valid contents here? I guess `always`, `madvise` and `never`.
But it seems also `[never]` is valid? Should we just check for `madvise` perhaps?